### PR TITLE
Adding community description in addition to sidebar, like site.

### LINKED ISSUE #5078: Separate `description` and `sidebar` fields for communities
### Requirements

- [X] Is this a feature request? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a feature request? Do not put multiple feature requests in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.
- [X] Do you agree to follow the rules in our [Code of Conduct](https://join-lemmy.org/docs/code_of_conduct.html)?

### Is your proposal related to a problem?

Currently, the instance sidebar is made up of two distinct `Site` fields - `description` and `sidebar`:

The shorter `description` field is used by instance-searching websites such as https://join-lemmy.org. But, unlike the instance sidebar, the community sidebar is made up of only one field. It would be nice if communities also had a two-field system like instances do. That way, a community search UI could render a short description about each result in a similar way to https://join-lemmy.org.

### Describe the solution you'd like.

Add separate `description` and `sidebar` fields to `Community`

### Describe alternatives you've considered.

It's currently possible to just render the first few lines of the full sidebar text, but this doesn't always produce desirable results due to the way community moderators choose to format the sidebar. Having a separate `description` field for a more concise description would be preferable imo.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dessalines**
- Also made changes to lemmy's group apub to be similar to its site, which uses content for the sidebar, and summary for the short description.
- Fixes #5078

@Nutomic when you get a chance, I could use your help with this peertube issue failing in tests: https://woodpecker.join-lemmy.org/repos/129/pipeline/9615/19#L282

I tried adding `deserialize_skip_error` to all the new fields I added, but it still is getting a parsing error.

### CODE REVIEW

File: crates/db_views_actor/src/community_view.rs [hunk 1]
```diff
@@ -123,6 +123,8 @@ fn queries<'a>() -> Queries<
       let searcher = fuzzy_search(&search_term);
       let name_filter = community::name.ilike(searcher.clone());
       let title_filter = community::title.ilike(searcher.clone());
+      // TODO this currently filters the new short description column, but should it also check the
+      // sidebar too now?
```
Comment (dessalines) [inline crates/db_views_actor/src/community_view.rs:127]: Not sure what should be done here, when they want to include the description filter. I could do either:

- Check all 4 (community name, title, description, and sidebar)
- Check 3 (name, title, description)

File: migrations/2024-10-16-141718_add_short_community_description/up.sql
Comment (dessalines) [inline migrations/2024-10-16-141718_add_short_community_description/up.sql:1]: These match exactly what was done to the site table when we extracted these into two also.

File: crates/db_views_actor/src/community_view.rs (hunk 1 above)
Comment (Nutomic) [reply crates/db_views_actor/src/community_view.rs:127]: Sidebar has a higher chance for false positives, as it includes all kinds of info about rules, related communities etc. So I would only check name, title and short description.

File: crates/db_views/src/post_view.rs [hunk 2]
```diff
@@ -1835,6 +1835,7 @@ mod tests {
         actor_id: inserted_community.actor_id.clone(),
         local: true,
         title: "nada".to_owned(),
+        sidebar: None,
         description: None,
         updated: None,
         banner: None,
```
Comment (Nutomic) [inline crates/db_views/src/post_view.rs:1841]: These tests would be simpler if you assert only specific fields, instead of building a full dummy object. Then its not necessary to make a change every time some unrelated field changes.

File: crates/apub/src/objects/community.rs [hunk 3]
```diff
@@ -107,8 +108,10 @@ impl Object for ApubCommunity {
       id: self.id().into(),
       preferred_username: self.name.clone(),
       name: Some(self.title.clone()),
-      summary: self.description.as_ref().map(|b| markdown_to_html(b)),
-      source: self.description.clone().map(Source::new),
+      content: self.sidebar.as_ref().map(|d| markdown_to_html(d)),
+      source: self.sidebar.clone().map(Source::new),
+      summary: self.description.clone(),
+      media_type: self.sidebar.as_ref().map(|_| MediaTypeHtml::Html),
```
Comment (Nutomic) [inline crates/apub/src/objects/community.rs:114]: This renaming will also cause some problems with backwards compat. But it does make more sense with the field definitions, so we probably need to do it this way (also for api).

File: crates/apub/src/protocol/objects/group.rs [hunk 4]
```diff
@@ -50,10 +51,16 @@ pub struct Group {
...
+  #[serde(deserialize_with = "deserialize_skip_error", default)]
+  pub(crate) content: Option<String>,
   #[serde(deserialize_with = "deserialize_skip_error", default)]
   pub(crate) source: Option<Source>,
   #[serde(deserialize_with = "deserialize_skip_error", default)]
+  pub(crate) media_type: Option<MediaTypeHtml>,
+  // short instance description
+  #[serde(deserialize_with = "deserialize_skip_error", default)]
+  pub(crate) summary: Option<String>,
   pub(crate) icon: Option<ImageObject>,
```
Comment (Nutomic) [inline crates/apub/src/protocol/objects/group.rs:64]: You accidentally removed the line `#[serde(deserialize_with = "deserialize_skip_error", default)]` for icon, thats why tests are failing.

File: crates/apub/src/objects/community.rs (hunk 3 above)
Comment (dessalines) [reply crates/apub/src/objects/community.rs:114]: > Also they will try to update description with text that is too long. As the field name and type are unchanged, this will also be hard to notice during testing. So for backwards compat it would be better to leave description unchanged, and use a new name for the short description.

Couldn't we truncate to account for that? To me having these fields named different things for instance and community would be a bit scatterbrained, and if we don't make them consistent now, we never will.

File: crates/apub/src/protocol/objects/group.rs (hunk 4 above)
Comment (dessalines) [reply crates/apub/src/protocol/objects/group.rs:64]: Aaahh dang, thx.

File: crates/apub/src/objects/community.rs (hunk 3 above)
Comment (Nutomic) [reply crates/apub/src/objects/community.rs:114]: Truncating would make sense, see https://github.com/LemmyNet/lemmy/issues/1375

File: crates/db_views/src/post_view.rs (hunk 2 above)
Comment (dessalines) [reply crates/db_views/src/post_view.rs:1841]: There's a lot of these, so we should probably do that later as part of a test cleanup.

File: crates/apub/src/objects/community.rs (hunk 3 above)
Comment (dessalines) [reply crates/apub/src/objects/community.rs:114]: Hrm... we should probably do these all at once then. So for all incoming federation receives, truncate on limited fields before inserting / updating?

Not sure if that should be done in this PR or a separate one.

Comment (Nutomic) [reply crates/apub/src/objects/community.rs:114]: Exactly, and better in a separate PR.

File: crates/db_views_actor/src/community_view.rs (hunk 1 above)
Comment (dessalines) [reply crates/db_views_actor/src/community_view.rs:127]: K cool, its good as is then. I'll just remove the comment.

**Comment by dessalines** [comment]
Okay this is also ready for review then.
